x/precisebank/types/genesis.go

```diff
@@ -1,17 +1,30 @@
 package types
 
-// Validate performs basic validation of supply genesis data returning an
-// error for any failed validation criteria.
+import "fmt"
+
+// Validate performs basic validation of genesis data returning an  error for
+// any failed validation criteria.
 func (gs *GenesisState) Validate() error {
+	if err := gs.Balances.Validate(); err != nil {
+		return fmt.Errorf("invalid balances: %w", err)
+	}
+
+	// TODO:
+	// - Validate remainder amount
+	// - Validate sum(fractionalBalances) + remainder = whole integer value
+	// - Cannot validate here: reserve account exists & balance match
```

Comment:
this only gets validated as a chain invariant, yes?

Reply:
The invariant can validate that, and we can validate other forms of the same validation during execution.

However, it seems like a good thing to test in InitGenesis? We can query the reserve balance on init to ensure the starting state of the module is valid.

Reply:
yup the last point we can validate in InitGenesis, just can't do it in this Validate() method since it needs some other keeper methods